src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx

```diff
@@ -84,20 +100,14 @@ const EditDomainDialog = ({
     <ActionForm
       initialValues={{
         forceAction: false,
-        domainName: domains.find(({ ethDomainId }) =>
-          selectedDomainId
-            ? ethDomainId.toString() === selectedDomainId
-            : ethDomainId !== ROOT_DOMAIN_ID,
+        domainName: domains.find(
+          ({ ethDomainId }) => ethDomainId === selectedDomainId,
```

Comment:
I am afraid I need to search once again to get the name. I can of course do it in different places but this will have to happen twice (also only when it's root or no domain). 

Maybe I'm not seeing something? I am open to your suggestions.

Reply:
what about something like that:
  const selectedDomain = useMemo(
    () => {
      const filteredDomains = domains.filter(({ethDomainId}) => ethDomainId !== ROOT_DOMAIN_ID);
      const domain = filteredDomains.find(({ethDomainId}) => ethDomainId === preselectedDomainId);
      return domain ? domain : filteredDomains[0];
    },
    [preselectedDomainId, domains],
  );

Then you at least dont have additional loop outside of memo function. Of course you would need to adjust it to the rest of things

Reply:
I think I had a better idea that we indeed only have one loop. Hope you'll like this option.